x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/index/IndexResolver.java

```diff
@@ -718,6 +725,10 @@ private static List<EsIndex> buildIndices(
             final InvalidMappedField invalidField = validityVerifier.apply(fieldName, types);
             // apply verification for fields belonging to index aliases
             Map<String, InvalidMappedField> invalidFieldsForAliases = getInvalidFieldsForAliases(fieldName, types, aliases);
+            // For ESQL there are scenarios where there is no field asked from field_caps and the field_caps response only contains
+            // the list of indices. To be able to still have an "indices" list properly built (even if empty), the metadata fields are
+            // accepted but not actually added to each index hierarchy.
+            var isMetadataField = allowedMetadataFields != null && allowedMetadataFields.contains(fieldName);
```

Comment:
This could be extracted into a method and also used in the loop above.

Reply:
Imo, the logic is too simple to warrant a separate method. Not-null set and a `contains`....

Reply:
But then why extract it as variable? :) It'd be small and inlined, code dedup'ing and easier to read (especially in the loop above). 
But bikeshedding, was just a suggestion, can stay as is.